x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlAuthenticator.java

```diff
@@ -221,6 +225,9 @@ private Assertion decrypt(EncryptedAssertion encrypted) {
                 }
             }
         }
+
+        warnOnReservedAttributeNames(assertion, attributes);
```

Comment:
Sanity check: we should be checking on all attributes, including encrypted ones, correct?